src/Mod/PartDesign/App/FeaturePad.cpp

```diff
@@ -214,12 +215,20 @@ App::DocumentObjectExecReturn *Pad::execute(void)
         if (!base.IsNull()) {
 //             auto obj = getDocument()->addObject("Part::Feature", "prism");
 //             static_cast<Part::Feature*>(obj)->Shape.setValue(getSolid(prism));
-            // Let's call algorithm computing a fuse operation:
-            BRepAlgoAPI_Fuse mkFuse(base, prism);
-            // Let's check if the fusion has been successful
-            if (!mkFuse.IsDone())
-                return new App::DocumentObjectExecReturn("Pad: Fusion with base feature failed");
-            TopoDS_Shape result = mkFuse.Shape();
+			TopoDS_Shape result;
+			if (getBaseObject()->getTypeId().isDerivedFrom(PartDesign::ShapeBinder::getClassTypeId()))

```

Comment:
I wonder, why didn't you modify getBaseObject? Because the issue will repeat itself on Revolution, and you fixed it only for Pad...

Reply:
Also, as a suggestion for improving the fix.

Why only ShapeBinder? What if something else pops up in the future? I would suggest to check for if Pad is housed by Body, and not return Sketch.Support from getBaseObject in the case. Because the whole point of getBaseFeature returning Sketch.Support is for legacy projects, and a body is a good sign the project is not legacy.

Reply:
You mean changing getBaseObject to return an empty shape, so that getBaseShape will throw an exception and a new base will be created by `base = TopoDS_Shape();` ?

I am not sure if I understood your point to check if pad is part of body. Could you give a little code example. I am new to this code and do not understand all of it.